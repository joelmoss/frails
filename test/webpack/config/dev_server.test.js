const createConfig = require('../../../package/webpack/create_config')

describe('config.devServer', () => {
  test('boolean', () => {
    expect(createConfig({ devServer: false })).toMatchSnapshot()
  })

  test('object', () => {
    expect(
      createConfig({ devServer: { hot: true, watchOptions: { poll: true } } })
    ).toMatchSnapshot()
  })
})
