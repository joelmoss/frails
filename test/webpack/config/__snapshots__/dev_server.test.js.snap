// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config.devServer boolean 1`] = `
Object {
  "entry": "./app/assets/application.js",
  "mode": "development",
  "output": Object {
    "path": "/Users/joelmoss/dev/frails/test/dummy/public/assets",
    "publicPath": "/assets/",
  },
}
`;

exports[`config.devServer object 1`] = `
Object {
  "devServer": Object {
    "contentBase": "/Users/joelmoss/dev/frails/test/dummy/public/assets",
    "hot": true,
    "port": 8080,
    "public": "localhost:8080",
    "publicPath": "/assets/",
    "watchOptions": Object {
      "ignored": "**/node_modules/**",
      "poll": true,
    },
  },
  "entry": "./app/assets/application.js",
  "mode": "development",
  "output": Object {
    "path": "/Users/joelmoss/dev/frails/test/dummy/public/assets",
    "publicPath": "/assets/",
  },
}
`;
